.github/workflows/cd.yml

```
@@ -28,44 +39,79 @@ jobs:
       fail-fast: false
       matrix:
         platform: [iOS, macOS]
-        uploadto: [app-store, ftp]
     runs-on: macos-13
 
     steps:
+
     - name: Checkout code
       uses: actions/checkout@v4
 
-    - name: Set iOS extra xcode params
-      if: matrix.platform == 'iOS'
-      run: echo "EXTRA_XCODEBUILD=-sdk iphoneos ${{ env.APPLE_AUTH_PARAMS }}" >> $GITHUB_ENV
+    - name: Install python dependencies
+      run: pip install pyyaml
 
-    - name: Set macOS FTP export method
-      if: matrix.platform == 'macOS' && matrix.uploadto == 'ftp'
-      run: echo "EXPORT_METHOD=developer-id" >> $GITHUB_ENV
+    - name: Set VERSION from code
+      shell: python
+      run: |
+        import yaml
+        import os
+        project = yaml.safe_load(open("project.yml"))
+        version = project['targets']['Kiwix']['settings']['base']['MARKETING_VERSION']
+        with open(os.getenv("GITHUB_ENV"), "a") as fh:
+            fh.write(f"VERSION={version}\n")
 
-    - name: Decide whether building nightly or release
-      env:
-        UPLOAD_TO: ${{ matrix.uploadto }}
+    - name: Get the date of last change
+      run: echo "DATE_OF_LAST_CHANGE=`git log -1 --format=%ct`" >> $GITHUB_ENV
+
+    - name: Check for code changes in the last week
       shell: python
       run: |
         import datetime
         import os
+        now = datetime.datetime.now()
+        last_change = ${{ env.DATE_OF_LAST_CHANGE }}
+        last_change = datetime.datetime.fromtimestamp(last_change)
```

remove previous line then
```suggestion
        last_change = datetime.datetime.fromtimestamp(int(os.getenv(DATE_OF_LAST_CHANGE)))
```
safer than to directly write the env thing in source code.